editing_tools/formatter.rb

```diff
@@ -0,0 +1,102 @@
+require 'tempfile'
+require 'pathname'
+require 'fileutils'
+require 'readline'
+require_relative 'rubima-lint'
+
+class RubimaFomatter
+  class BaseFormatter
+    include RubimaLint::Rules
+    
+    def initialize(line_no, line)
+      @line_no = line_no
+      @line = line
+    end
+
+    def apply
+      pre_checked_line = pre_check
+      return @line if @line == pre_checked_line
+
+      puts "#{formated_message}: line [#{@line_no}]:"
+      puts pre_checked_line
+
+      loop do
+        y_or_n = Readline.readline("appry? [y(es)<default>|n(o)|e(dit)|i(gnore)]> ")
+        case y_or_n.chomp
+        when ?y, 'yes', ''
+          return convert_line
+        when ?n, 'no'
+          return @line
+        when ?e, 'edit'
+          return edit_line
+        end
+      end
+    end
+
+    def edit_line
+      editor = ENV['EDITOR'] || 'vim'
+      tempfile = Tempfile.open { |fp| fp.print @line; fp }
+      system [editor, tempfile.path].join(' ')
+      File.read(tempfile.path)
+    end
+  end
+
+  class WhiteSpaceFormatter < BaseFormatter
+    def formated_message
+      "半角文字列の前後に空白を挿入しました"
+    end
+
+    def pre_check
+      convert_line("\e[7m \e[m")
+    end
+
+    def convert_line(blank = ' ')
+      @line.gsub(MISSING_BLANK, blank)
+    end
+  end
+
+  class TrimUnnecessarySpaceFormatter < BaseFormatter
+    def formated_message
+      "不要な空白を削除しました"
+    end
+
+    def pre_check
+      convert_line("\e[32m#{$&}\e[m")
+    end
+
+    def convert_line(trimed = '')
```

Comment:
```suggestion
    def convert_line(trimmed = '')
```
https://github.com/rubima/magazine.rubyist.net/pull/440#issuecomment-1581582284